### DISCUSSION TITLE
Custom version of ansible-form

### DISCUSSION THREAD

**Discussion by infinite1by0** [ansibleguy76/ansibleforms#149] [Q&A] [answered]
Hey I wish to modify AF to have my own custom version of it, I want to edit home page, some navbar etc. Can anyone guide me how doable this is, I havent worked on Vuejs. Where can I start from. Thankyou

**Comment by ansibleguy76** [comment]
Hi
The logo is already customizable. 
color is probably also doable. 

what about the navbar do you want to change?

**Reply by infinite1by0** [reply]
Yeah I got that,
I want to hide some nav elements like change password,/ add customized nav links, also I have only one form so i  want to edit home page so that it either directly shows that form, or automatically redirect the /#/ route to /#/form/?form=myformname. I know these things require modifying AF code. Right now i am running AF using docker-compose, How to modify the AF and use updated version using compose, thankyou

**Reply by ansibleguy76** [reply]
make a list and i will see what i can do. hiding a few options with an env variable would be simpel and i could auto redirect to a form if there is only 1 form present.

**Reply by infinite1by0** [reply]
Sure
0. How to change navbar color

1. Option to have hide any particular navlink
2. Option to automatically redirect /#/ to a desired form /#/form/?form=myformname, Actually this need not only happen when we have only one form, (what i was thinking is we have a landing form, instead of default /#/ from there we can go to different form using hyperlinks)
for eg-
![image](https://github.com/ansibleguy76/ansibleforms/assets/91357272/ffe68e99-3504-4372-a5bb-cef7ac9da969)
here in /client/src/router.js maybe in `path:"/"` if we could have a option which if set will `redirect: '/#/form/?form=myformname'` to a defined path instead of rendering the component Forms

Although this is small task please let me know if you can have these option in AF, Thankyou

**Reply by infinite1by0** [reply]
Hi, Is there any plans regarding above option( specially 1st one) ?

**Reply by ansibleguy76** [reply]
this wont be soon.  i want to focus on actual bugs and features.  this is more a nice to have, almost custom changes for you.

**Reply by infinite1by0** [reply]
I totally understand that, can we modify AF code and use the modified version docker image in our docker-compose file, it would be great if you could help me in this. How do i create new docker image? like is there any dockerfile? Thanks

**Comment by ansibleguy76** [comment, marked as answer]
in the code there is indeed a docker-file.  
https://github.com/ansibleguy76/ansibleforms/blob/main/server/Dockerfile
